### PR TITLE
Refactor pointer interaction tests - Pass 1

### PR DESCRIPTION
**Pull request by aicioara**
Part of #2628

I will be doing 2 passes because I will get more context (and better sense of best practices) by refactoring other files. However, please flag anything that seems unusual.

As part of this: 
- **Test semantics: hierarchy, naming, testing, but no coverage check**
- Applied best practices from the wiki page
- Factored out common code into `beforeEach()` clauses
- No nested `beforeEach()`es
- Cleaned the console
- Remade hierarchy as shown bellow
screen shot 2015-09-11 at 4 33 29 pm

### CODE REVIEW

File: test/interactions/pointerInteractionTests.ts [hunk 1]
```diff
@@ -1,230 +1,220 @@
...
-    it("onPointerExit", () => {
-      let svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
-      let c = new Plottable.Component();
-      c.renderTo(svg);
+        eventTarget = component.background();
 
-      let pointerInteraction = new Plottable.Interactions.Pointer();
-      pointerInteraction.attachTo(c);
+        callbackCalled = false;
+        callback = function(point) {
```
Comment (jtlan) [inline test/interactions/pointerInteractionTests.ts]: Fat-arrow notation can be used here.

File: test/interactions/pointerInteractionTests.ts [hunk 2]
```diff
@@ -1,230 +1,220 @@
...
+        TestMethods.triggerFakeMouseEvent("mousemove", eventTarget, insidePoint.x, insidePoint.y);
+        TestMethods.triggerFakeMouseEvent("mousemove", eventTarget, outsidePoint.x, outsidePoint.y);
+
+        assert.isFalse(enterCallback1Called, "callback 1 was disconnected from pointer enter interaction");
+        assert.isTrue(enterCallback2Called, "callback 2 is still connected to the pointer enter interaction");
+
+        svg.remove();
+      });
+
+      it("calls all the drag pointer interaction callbacks when needed", () => {
```
Comment (jtlan) [inline test/interactions/pointerInteractionTests.ts]: What's this test trying to test for? Everything inside this `it()` seems to be tested elsewhere already.

Comment (aicioara) [inline test/interactions/pointerInteractionTests.ts]: It's being tested that the interactions behave well when interacting to each other and that one is not overriding the others. However I am not sure about the usefulness of this and it also takes a significant amount of time. Should I delete it? 

The inspiration for it comes from other tests (DragLayerTests, Drag Interaction)

Comment (jtlan) [inline test/interactions/pointerInteractionTests.ts]: It seems like it's trying to test the independence of each event. While that's something that should work, it doesn't seem scalable to test it (O(n^2) in the number of events...).

Comment (aicioara) [inline test/interactions/pointerInteractionTests.ts]: will remove

**Pull request merged**